rules/prefer-component.js

```diff
@@ -0,0 +1,73 @@
+/*
+ * Since AngularJS 1.5, we can use a new API when creating directives.
+ * This new API should be use when creating directive without DOM manipulation.
+ *
+ * @version 0.16.0
+ * @category bestPractice
+ * @sinceAngularVersion 1.5
+ */
+'use strict';
+
+var angularRule = require('./utils/angular-rule');
+
+module.exports = angularRule(function(context) {
+    var potentialReplaceNodes = {};
+
+    function addPotentialLinkNode(variableName, node) {
+        var nodeList = potentialReplaceNodes[variableName] || [];
+
+        nodeList.push({
+            name: variableName,
+            node: node,
+            block: context.getScope().block.body
+        });
+
+        potentialReplaceNodes[variableName] = nodeList;
+    }
+
+    return {
+        'angular:directive': function(callExpressionNode, fnNode) {
+            if (!fnNode || !fnNode.body) {
+                return;
+            }
+            fnNode.body.body.forEach(function(statement) {
+                if (statement.type === 'ReturnStatement' && !potentialReplaceNodes[statement.argument.name || '']) {
+                    context.report(statement, 'Directive should be implemented with the component method.');
+                }
+            });
+        },
+        AssignmentExpression: function(node) {
+            // Only check for literal member property assignments.
+            if (node.left.type !== 'MemberExpression') {
+                return;
+            }
+            // Only check setting properties named 'link'.
+            if (node.left.property.name !== 'link') {
```

Comment:
Shouldn't `compile`, `multiElement`, `priority`, `templateNamespace`,  `terminal` and `restrict` allowing `A` or `C` be allowed as well? (See https://docs.angularjs.org/api/ng/service/$compile#comprehensive-directive-api and https://docs.angularjs.org/guide/component#comparison-between-directive-definition-and-component-definition)